R/interface.R

```diff
@@ -79,18 +79,26 @@
 ##'   variable ordering, or any processing done while accepting the
 ##'   data object used to create the data fed into the particles.
 ##'
-##' @title Create a dust model from a C++ input file
+##' @section Confuring your model:
 ##'
-##' @param filename The path to a single C++ file
+##' You can optionally use C++ psuedo-attributes to configure the
+##'   generated code. Currently we support two attributes:
+##'
+##' * `[[dust::class(classname)]]` will tell dust the name of your
+##'   target C++ class (in this example `classname`). You will need to
+##'   use this if your file uses more than a single class, as
+##'   otherwise will try to detect this using extremely simple
+##'   heuristics.
 ##'
-##' @param type The name of the "type" (the C++ class) that represents
-##'   your model.  If \code{NULL} we try to work this out from your
-##'   file using extremely simple heuristics.
+##' * `[[dust::name(modelname)]]` will tell dust the name of the model
+##'   for exporting to R. For technical reasons this must be
+##'   alphanumeric characters only (sorry, no underscore) and must not
```

Comment:
Don't need to do this now, but in either #98 or odin.dust docs I think it's worth explicitly stating that underscores aren't allowed (as you do here). Particularly in model filenames, as I think that will be a common issue!

Reply:
There will be an odin.dust update following this, so I can do that there